src/lightning/pytorch/utilities/model_summary/model_summary.py

```diff
@@ -120,8 +119,13 @@ def layer_type(self) -> str:
     @property
     def num_parameters(self) -> int:
         """Returns the number of parameters in this module."""
-        return sum(
-            cast(int, np.prod(p.shape)) if not _is_lazy_weight_tensor(p) else 0 for p in self._module.parameters()
+        return int(
```

Comment:
why do we need to cast `int` again if sum values are already `int`

Reply:
It returns a tensor as Borda mentioned in the above conversation, but in the doctest it returns an int. Hence I made it int manually.

Reply:
I have updated the code and removed the cast to `int`